### PR TITLE
Update ConfigurableEncryptor.java

### PR DESCRIPTION
**Pull request by YYTVicky**
(It is very **important** that you created an Apache Knox JIRA for this change and that the PR title/commit message includes the Apache Knox JIRA ID!)

## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. For instance: running automated unit/integration tests, manual tests. Please write down your test steps as detailed as possible)
(If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

Please review [Knox Contributing Process](https://cwiki.apache.org/confluence/display/KNOX/Contribution+Process#ContributionProcess-GithubWorkflow) before opening a pull request.

### CODE REVIEW

**Comment by YYTVicky** [comment]
Hi, 

  Could I know why it does not pass the test, I try to leave a comment message to inform that it might be better not to use a constant password for the invoked function.  

Thanks!

**Comment by moresandeep** [comment]
> Hi,
> 
> Could I know why it does not pass the test, I try to leave a comment message to inform that it might be better not to use a constant password for the invoked function.
> 
> Thanks!

You have check style failures 
```
[ERROR] /home/runner/work/knox/knox/gateway-spi/src/main/java/org/apache/knox/gateway/services/security/impl/ConfigurableEncryptor.java:110: Line matches the illegal pattern 'Trailing whitespace'. [Regexp]
[ERROR] /home/runner/work/knox/knox/gateway-spi/src/main/java/org/apache/knox/gateway/services/security/impl/ConfigurableEncryptor.java:111: Line matches the illegal pattern 'Trailing whitespace'. [Regexp]
[ERROR] /home/runner/work/knox/knox/gateway-spi/src/main/java/org/apache/knox/gateway/services/security/impl/ConfigurableEncryptor.java:113: Line matches the illegal pattern 'Trailing whitespace'. [Regexp]
Audit done.
[INFO] There are 3 errors reported by Checkstyle 8.31 with build-tools/checkstyle/checker.xml ruleset.
[ERROR] src/main/java/org/apache/knox/gateway/services/security/impl/ConfigurableEncryptor.java:[110] (regexp) Regexp: Line matches the illegal pattern 'Trailing whitespace'.
[ERROR] src/main/java/org/apache/knox/gateway/services/security/impl/ConfigurableEncryptor.java:[111] (regexp) Regexp: Line matches the illegal pattern 'Trailing whitespace'.
[ERROR] src/main/java/org/apache/knox/gateway/services/security/impl/ConfigurableEncryptor.java:[113] (regexp) Regexp: Line matches the illegal pattern 'Trailing whitespace'.
```
see: https://github.com/apache/knox/pull/327/checks?check_run_id=616662599

**Comment by YYTVicky** [comment]
The parameter "tmp.getencoded()" used in  SecretKeySpec should not be a constant value,  our tool suggest following new fix patch for   
**SecretKey secret = new SecretKeySpec(tmp.getEncoded(), alg);**

KeyGenerator keyGen = KeyGenerator.getInstance("AES");
keyGen.init(128);
SecretKey secretKey = keyGen.generateKey();
 byte[] tmp= secretKey.getEncoded();
SecretKey secret = new SecretKeySpec(tmp, alg);

pls kindly take a look and hope to get feedback from you!

File: gateway-spi/src/main/java/org/apache/knox/gateway/services/security/impl/ConfigurableEncryptor.java [hunk 1]
```diff
@@ -107,6 +107,14 @@ public EncryptionResult encrypt(byte[] plain) throws Exception {
...
 
     SecretKey tmp = getKeyFromPassword(new String(passPhrase), salt);
+    /*
+    * the key should not be constant value, recommend code:
+    * SecureRandom random = new SecureRandom();
+    * String defaultKey = String.valueOf(random.nextInt());
+    * byte[] keyBytes = defaultKey.getBytes();
+    * keyBytes = Arrays.copyOf(keyBytes,24);
+    * SecretKey secret = new SecretKeySpec(keyBytes, alg);
+    */
```
Comment (risdenk) [inline gateway-spi/src/main/java/org/apache/knox/gateway/services/security/impl/ConfigurableEncryptor.java:117]: This is a comment and not actually fixing anything.

**Comment by risdenk** [comment]
Closing since no update in ~30 days.

**Pull request closed without merging**